uobtheatre/productions/test/test_models.py

```diff
@@ -678,14 +678,35 @@ def test_performance_str(name, start, string):
 
 
 @pytest.mark.django_db
-def test_performance_min_price():
+def test_performance_min_price_free_performance():
 
+    performance = PerformanceFactory()
+    PerformanceSeatingFactory(performance=performance, price=0)
+    PerformanceSeatingFactory(performance=performance, price=20)
+
+    assert performance.min_seat_price() == 0
+
+
+@pytest.mark.django_db
+def test_performance_min_price_with_single_discounts():
     performance = PerformanceFactory()
     PerformanceSeatingFactory(performance=performance, price=30)
     PerformanceSeatingFactory(performance=performance, price=10)
-    PerformanceSeatingFactory(performance=performance, price=20)
 
-    assert performance.min_seat_price() == 10
+    adult = ConcessionTypeFactory(name="adult")
+    child = ConcessionTypeFactory(name="child")
+
+    null_discount = DiscountFactory(percentage=0, name="adult")
+    DiscountRequirementFactory(discount=null_discount, number=1, concession_type=adult)
+
+    single_discount = DiscountFactory(percentage=0.27, name="child")
+    DiscountRequirementFactory(
+        discount=single_discount, number=1, concession_type=child
+    )
+    null_discount.performances.set([performance])
+    single_discount.performances.set([performance])
+
+    assert performance.min_seat_price() == math.ceil((1 - 0.27) * 10)
```

Comment:
Personally I think these kind of assertions should be explicit and use as little function/complication as possible. Having code and not absolutes in assertions makes it harder to understand unit tests like these, as well as introduces the possibility the the output it not actually what you want (because you just copied the logic from the function in the model).

```suggestion
    assert performance.min_seat_price() == 8
```